### PR TITLE
test(e2e): add e2e tests for page footer platform

### PR DESCRIPTION
**Pull request by puru-hk**
#### Please provide a link to the associated issue.
Closes

#### Please provide a brief summary of this pull request.
add e2e tests for page footer component (Platform)

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [x] update `README.md`:N/A
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes) :N/A
- [x] Documentation Examples: N/A
- [x] Stackblitz works for all examples: N/A

### CODE REVIEW

**Comment by Seamoo13** [comment]
please get a passing Travis build before requesting reviews

File: e2e/wdio/platform/tests/page-footer.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            for (let i = 0; i < linksLength; i++) {
+                scrollIntoView(pagefooterclicablelink, i);
+                expect(isElementClickable(pagefooterclicablelink, i)).toBe(true, 'link with index ${i} not clickable');
+            }
+        });
+
+        it('should check pagefooter text', () => {
+            expect(getText(pagefootertext,)).toEqual('1992-2021 @copyright all right reserved.', 'Text is not matching Index');
+            expect(getText(pagefootertext, 1)).toEqual('1992-2021 @copyright all right reserved.', 'Text is not matching Index1');
+            expect(getText(pagefootertext, 2)).toEqual('1992-2021 @copyright all right reserved.', 'Text is not matching Index2');
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/page-footer.e2e-spec.ts:29]: when the assertion is clear (expect to equal 'this specific text') then you don't need to add a failure message. We only do that if the assertion is true/false (expect to be true) to add clarity as to what should be happening.

File: e2e/wdio/platform/tests/page-footer.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        });
+
+    describe('Check visual regression basic', function() {
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/page-footer.e2e-spec.ts:32]: please add a skip, as discussed on the previous week's call, we have temporarily skipped visual regression cases until we get a new tool up, due to instability and amount of time QA spends daily on updating screenshots in multiple PRs. to do this, just rename the `describe` to `xdescribe`

File: e2e/wdio/platform/tests/page-footer.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            expect(pagefooterPage.compareWithBaseline()).toBeLessThan(5);
+        });
+
+    describe('Check orientation', function() {
+        it('Verify RTL and LTR orientation', () => {
+            pagefooterPage.checkRtlSwitch();
+            });
+        });
+    });
+});
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/page-footer.e2e-spec.ts:45]: just a general note, when the test spec is so small, you can organize it 2 ways which are a little better for reading.

way 1.

describe(x test suite)..
describe('main checks')..
it..
it..
it..
describe('visual regression and RTL)..
it..
it..



way 2.

describe(x test suite)..
it..
it..
it..
it..

File: e2e/wdio/platform/tests/page-footer.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import { PageFooterPo } from '../pages/page-footer.po';
+import {
+    getElementArrayLength,
+    getText,
+    isElementClickable,
+    scrollIntoView,
+} from '../../driver/wdio';
+
+describe('PageFooter test suite:', function() {
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/page-footer.e2e-spec.ts:9]: page footer is 2 words

File: e2e/wdio/platform/tests/page-footer.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    
+        it('verify clickable links ', () => {
+            const linksLength = getElementArrayLength(pagefooterclicablelink);
+            for (let i = 0; i < linksLength; i++) {
+                scrollIntoView(pagefooterclicablelink, i);
+                expect(isElementClickable(pagefooterclicablelink, i)).toBe(true, 'link with index ${i} not clickable');
+            }
+        });
+
+        it('should check pagefooter text', () => {
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/page-footer.e2e-spec.ts:26]: page footer is 2 words

File: e2e/wdio/platform/tests/page-footer.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+describe('PageFooter test suite:', function() {
+    const pagefooterPage = new PageFooterPo();
+    const {
+        pagefooterclicablelink, pagefootertext  } = pagefooterPage;
+
+    beforeAll(() => {
+        pagefooterPage.open();
+    }, 1);
+    
+        it('verify clickable links ', () => {
```
Comment (Seamoo13) [inline e2e/wdio/platform/tests/page-footer.e2e-spec.ts:18]: this sounds like you are only checking a certain set of links. please rename to 'verify links are clickable'

File: e2e/wdio/platform/pages/page-footer.po.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import { waitForPresent } from '../../driver/wdio';
+
+export class PageFooterPo extends BaseComponentPo {
+    private url = '/page-footer';
+    pagefooterclicablelink = '.fd-page-footer__container fdp-link';
+    pagefootertext = '.fd-page-footer__text .fd-form-label';
+    
+    open(): void {
+        super.open(this.url);
+        waitForPresent(this.title);
```
Comment (Seamoo13) [inline e2e/wdio/platform/pages/page-footer.po.ts:11]: please also add `waitForElDisplayed(this.root);` here

**Review by Seamoo13 — CHANGES_REQUESTED** [review]

**Comment by Seamoo13** [comment]
also need to add url to the wdio-cron.config.js file or the cron job will not run tests here

**Comment by maxym-valor** [comment]
pls, fix tabulation for iterations. How it should be: 
image

File: e2e/wdio/platform/tests/page-footer.e2e-spec.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+    beforeAll(() => {
+        pageFooterPage.open();
+    }, 1);
+    
+        it('verify links are clickable', () => {
+            const linksLength = getElementArrayLength(pageFooterClickableLink);
+            for (let i = 0; i < linksLength; i++) {
+                scrollIntoView(pageFooterClickableLink, i);
+                expect(isElementClickable(pageFooterClickableLink, i)).toBe(true, 'link with index ${i} not clickable');
```
Comment (maxym-valor) [inline e2e/wdio/platform/tests/page-footer.e2e-spec.ts:26]: error message will not work as you wish, if you want to use template literal - quotes should be ``, not ''

File: e2e/wdio/platform/tests/page-footer.e2e-spec.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
+import { PageFooterPo } from '../pages/page-footer.po';
+import {
+    getElementArrayLength,
+    getText,
```
Comment (maxym-valor) [inline e2e/wdio/platform/tests/page-footer.e2e-spec.ts:4]: remove unused method

File: e2e/wdio/platform/pages/page-footer.po.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+export class PageFooterPo extends BaseComponentPo {
+    private url = '/page-footer';
+    pageFooterClickableLink = '.fd-page-footer__container fdp-link';
+    pageFooterText = '.fd-page-footer__text .fd-form-label';
+    
+    open(): void {
+        super.open(this.url);
+        waitForElDisplayed(this.root);
+    }
```
Comment (maxym-valor) [inline e2e/wdio/platform/pages/page-footer.po.ts:12]: why did you remove `waitForPresent(this.title);` ?

File: e2e/wdio/platform/tests/page-footer.e2e-spec.ts [hunk 11]
```diff
@@ -23,7 +22,7 @@ describe('Page Footer test suite:', function() {
             const linksLength = getElementArrayLength(pageFooterClickableLink);
             for (let i = 0; i < linksLength; i++) {
                 scrollIntoView(pageFooterClickableLink, i);
-                expect(isElementClickable(pageFooterClickableLink, i)).toBe(true, 'link with index ${i} not clickable');
+                expect(isElementClickable(pageFooterClickableLink, i)).toBe(true, "link with index ${i} not clickable");
```
Comment (maxym-valor) [inline e2e/wdio/platform/tests/page-footer.e2e-spec.ts:25]: wrong quotes again 😁
not "" and not '', should be ``
` expect(isElementClickable(pageFooterClickableLink, i)).toBe(true, `link with index ${i} not clickable`);`

**Review by Seamoo13 — APPROVED** [review]
Approved, but still need to make the formatting/spacing correction described by Max.

**Review by maxym-valor — APPROVED** [review]

**Review by Smiranin — APPROVED** [review]

**Pull request merged**